tool/teleport/common/sftp.go

```diff
@@ -89,10 +96,31 @@ type allowedOps struct {
 type sftpHandler struct {
 	logger  *log.Entry
 	allowed *allowedOps
-	events  chan<- *apievents.SFTP
+	files   []*trackedFile
+	events  chan<- protoAuditEvent
+}
+
+type trackedFile struct {
+	*os.File
```

Comment:
I would not embed this. If someone calls the underlying Read or Write methods won't we fail to update the counters?

Reply:
`sftp.Handlers` which we implement only returns `io.ReaderAt`s and `io.WriterAt`s, so `Read` and `Write` can't be called. `sftp.Handlers` does close the reader/writer if it implements `io.Closer`, but doesn't attempt to call anything else. So this should be safe.

Reply:
Good point, I removed the embedding so only the needed methods are exposed.